### ISSUE TITLE
change colours through shared colour styles

### ISSUE THREAD

**Issue by creativechris08** [themainingredient/colormate#83]
are you able to find and replace colors that are based on shared color styles (libraries) ive been after a plugin like this for over 2 years? :)

**Comment by SimoneFace** [comment]
Hey @creativechris08, thanks for the feedback!

If for example you want to update "_Tertiary Dark_" in your styles here using Colormate, this is what you would need to do:

**- Check what colour Tertiary Dark is in your styles, for argument sake let's say it's pure black #000000
- Then run Colormate on a selection or whole document.
- Replace the #000000 in Colormate with whatever colour you like, see it update on those objects in real time. 
- Go to your styles and you'll see an asterisk icon next to your "_Tertiary Dark_" style, click on this and select "update layer style". This will then implement your new replaced colour in your sketch styles.**   

I hope this answers your question? If you have any better suggestions on how Colormate might handle this we would be very happy to hear it and look at implementing in our roadmap.

Thanks
The Colormate Team

**Comment by creativechris08** [comment]
Hello thanks for the reply unfortunately the above method will not work for shared layer styles? 

My shared colour layer styles are in a separate shared sketch library and there is no way for the colourmate to identify them on my shared component layer overrides :( would be great if this was a feature: search and replace all shared colour layer styles throughout the document 👍🏻😎

**Comment by SimoneFace** [comment]
@creativechris08 ah I see, so your styles are in a library file that's also shared by other team mates? So you would want Colormate to find these styles and replace, presumably in your library file then too? 🤔

**Comment by creativechris08** [comment]
@SimoneFace if this would be possible you would be a legend, ive been waiting for this plugin for over two years, contacted sketch team and many others dev's but its never been achieved, not sure if there is API limitations; but the closest plugin ive found is overideit,, but this is still resrictive as you cant change every shared colour in one go (only same symbols)

images attached below is my abstract / sketch library set up (shared colour file to all brands)

<img width="2549" alt="Screenshot 2019-08-14 at 11 37 05" src="https://user-images.githubusercontent.com/43187049/63011386-7c321f00-be88-11e9-9540-30271d66c767.png">
<img width="2560" alt="Screenshot 2019-08-14 at 11 37 54" src="https://user-images.githubusercontent.com/43187049/63011394-80f6d300-be88-11e9-968a-261eec435b25.png">
<img width="460" alt="Screenshot 2019-08-14 at 11 38 04" src="https://user-images.githubusercontent.com/43187049/63011405-848a5a00-be88-11e9-82b4-1af2cb0c94ad.png">
<img width="2558" alt="Screenshot 2019-08-14 at 11 38 57" src="https://user-images.githubusercontent.com/43187049/63011411-881de100-be88-11e9-90e8-9332662f38bb.png">

**Comment by SimoneFace** [comment]
@creativechris08 Legendary status is extremely tempting! Thanks for taking the time to share the screenshots, it really helps us figure out next steps. I'll have a chat with the team and see if this is something we might consider in our next releases. We'll keep you posted! Anything else that occurs to you please feel free to share here and we'll consider it too. I can definitely see the appeal in sorting this out, we also use abstract and sketch for our work so it would make sense for Colormate to accommodate this shared setup. 💪

**Comment by creativechris08** [comment]
would be great to search for shared colour styles and to replace with another shared colour style,, this would replace the colour throughout the whole page / document ect :)

**Comment by creativechris08** [comment]
😎🎨🎨🎨😎🙏

**Comment by SimoneFace** [comment]
Ok sweet, thanks for sharing your setup that's going to massively help with us figuring out if we can do this with the API. We'll be in touch over the coming weeks😉 Thanks @creativechris08 !

**Comment by ghost** [comment]
Cool idea, I would also recommend splitting those shared styles into libraries and using Abstract or a plugin to swap libraries. The files where they're used would automatically update after accepting the library updates prompt. I'm using this method to change our platform app sketch files to cater for separate clients, changing everything from colour, buttons, radius, to product images.

**Comment by SimoneFace** [comment]
@lmjabreu-poq thanks for the feedback, every bit helps! So would your expectation be that Colormate could update style colours with other style colours? And also update a style colour by just selecting a new colour that is not already a part of a library? This way if you have your libraries set up a certain way with abstract etc it will update across all instances that the shared colour is applied on?

**Comment by SimoneFace** [comment]
@creativechris08 and @lmjabreu-poq would you guys have 2 minutes to fill out a brief survey to helps us develop version 2.0? 😊 We'd really appreciate your insight to make Colormate as awesome and as useful as we can to our users 🙏 https://themainingredient.typeform.com/to/uirGl4?source=github

**Comment by ghost** [comment]
> @lmjabreu-poq thanks for the feedback, every bit helps! So would your expectation be that Colormate could update style colours with other style colours? And also update a style colour by just selecting a new colour that is not already a part of a library? This way if you have your libraries set up a certain way with abstract etc it will update across all instances that the shared colour is applied on?

Yeah I'd say that's also a common use-case – replace all applications of a shared style with another, similar to [symbol swapper](https://github.com/sonburn/symbol-swapper). Like I said, my most common use-case is to replace **all** shared styles/symbols by swapping libraries, I'm doing this by using Abstract's library re-mapping (used to use a library swapper plugin, can't recall which).
